### PR TITLE
ceph: update cephobjectstore monitoring logic

### LINKED ISSUE #6186: cephobjectstore monitoring goroutine must always be stopped
Rework this logic since it does not handle the following flow:

			1. objectstore is deleted
			2. go routine is stopped
		 	3. objectstore is created
		 	4. objectstore is deleted
			5. the bool 'clusterResourceDeleted' is still true, so the running goroutine is never stopped

Here: https://github.com/rook/rook/blob/master/pkg/operator/ceph/object/controller.go#L226

### PR DESCRIPTION
**Pull request by subhamkrai**
currently, cephobjectstore monitoring is never
stopped for the following flow:
 ```
1. objectstore is deleted
 2. go routine is stopped
 3. objectstore is created
 4. objectstore is deleted
```
so, removing clusterResourceDeleted flag and
working with the map only.

Signed-off-by: subhamkrai <subhamkumarrai03@gmail.com>

**Description of your changes:**

**Which issue is resolved by this Pull Request:**
Resolves #6186

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/master/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [ ] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.

### CODE REVIEW

**Comment by Madhu-1** [comment]
Have you tried below scenario

* create ceph cluster
* create ceph object store
* delete ceph cluster
* delete ceph object store

**Comment by subhamkrai** [comment]
> Have you tried below scenario
> 
> * create ceph cluster
> * create ceph object store
> * delete ceph cluster
> * delete ceph object store

No, I have tested the above scenario only(in PR message).

File: pkg/operator/ceph/object/controller.go [hunk 1]
```diff
@@ -254,9 +253,6 @@ func (r *ReconcileCephObjectStore) reconcile(request reconcile.Request) (reconci
 			if err != nil {
 				return reconcile.Result{}, errors.Wrapf(err, "failed to delete store %q", cephObjectStore.Name)
 			}
```
Comment (leseb) [inline pkg/operator/ceph/object/controller.go:255]: I think we should move this block https://github.com/rook/rook/pull/6208/files#diff-7e97b37c53b4739ef8382f7b481ab8dcR240-R244 after `cfg.deleteStore()`, if `cfg.deleteStore()` fails we want to retry again, if we close the channel and remove from the map before then we will never re-enter this block.

**Review by leseb — CHANGES_REQUESTED** [review]

File: pkg/operator/ceph/object/controller.go (hunk 1 above)
Comment (subhamkrai) [reply pkg/operator/ceph/object/controller.go:249]: Thanks for the suggestion, Done.

Comment (leseb) [reply pkg/operator/ceph/object/controller.go:249]: Did you validate again with this new change?

Comment (subhamkrai) [reply pkg/operator/ceph/object/controller.go:249]: > Did you validate again with this new change?
Yes, I validate the latest changes.

**Review by leseb — APPROVED** [review]

**Pull request merged**